e2e/affinities/affinities.go

```diff
@@ -0,0 +1,228 @@
+package affinities
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/e2e/framework"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/jobspec"
+	"github.com/stretchr/testify/require"
+
+	. "github.com/onsi/gomega"
+)
+
+type BasicAffinityTest struct {
+	framework.TC
+	jobIds []string
+}
+
+func init() {
+	framework.AddSuites(&framework.TestSuite{
+		Component:   "Affinity",
+		CanRunLocal: true,
+		Cases: []framework.TestCase{
+			new(BasicAffinityTest),
+		},
+	})
+}
+
+func (tc *BasicAffinityTest) TestSingleAffinities(f *framework.F) {
+	nomadClient := tc.Nomad()
+
+	// Parse job
+	job, err := jobspec.ParseFile("affinities/input/aff1.nomad")
+	require := require.New(f.T())
+	require.Nil(err)
+	uuid := uuid.Generate()
+	jobId := helper.StringToPtr("aff" + uuid[0:8])
+	job.ID = jobId
+
+	tc.jobIds = append(tc.jobIds, *jobId)
+
+	// Register job
+	jobs := nomadClient.Jobs()
+	resp, _, err := jobs.Register(job, nil)
+	require.Nil(err)
+	require.NotEmpty(resp.EvalID)
+
+	g := NewGomegaWithT(f.T())
+
+	// Wrap in retry to wait until placement
+	g.Eventually(func() []*api.AllocationListStub {
+		// Look for allocations
+		allocs, _, _ := jobs.Allocations(*job.ID, false, nil)
+		return allocs
+	}, 5*time.Second, time.Second).ShouldNot(BeEmpty())
+
+	jobAllocs := nomadClient.Allocations()
+
+	allocs, _, _ := jobs.Allocations(*job.ID, false, nil)
+
+	// Verify affinity score metadata
+	for _, allocStub := range allocs {
+		alloc, _, err := jobAllocs.Info(allocStub.ID, nil)
+		require.Nil(err)
+		require.NotEmpty(alloc.Metrics.ScoreMetaData)
+		// Expect node affinity score to be 1.0 (dc1 nodes) if found
+		// TODO(preetha) make this more realistic
+		for _, sm := range alloc.Metrics.ScoreMetaData {
+			score, ok := sm.Scores["node-affinity"]
+			if ok {
```

Comment:
Should the test fail if !ok?

Reply:
not every node in the score metadata will match the affinity, so this is correct.